### ISSUE TITLE
windows_package resource bug: implicit argument passing of super from method defined by define_method() is not supported

### ISSUE THREAD

**Issue by cookiecurse** [chef/chef#9770]
## Description
Earlier in April the windows_package resource worked without issues, on 4/29 when I ran a cookbook I got the error below. Decided to test it standalone with an example from the docs and got the same error.

Seems like there was an update done to this resource 7days ago, but I am not 100% sure it is related:
https://github.com/chef/chef/commit/6ed90220893285f80513c6155620ebd0d2fbda1e

## Chef Version
Running this test in Kitchen (v1.25.0)
System Info:
       ------------
       chef_version=16.0.257
       platform=windows
       platform_version=10.0.14393
       ruby=ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x64-mingw32]
       program_name=C:/opscode/chef/bin/chef-client
       executable=C:/opscode/chef/bin/chef-client

## Replication Case
`reboot 'now' do`
`  action :nothing`
`  reason 'Need to reboot because SSMS is finished.'`
`  delay_mins 5`
`end`

`windows_package '7zip' do`
`  source 'http://www.7-zip.org/a/7z938-x64.msi'`
`  action :install`
`  notifies :request_reboot, 'reboot[now]', :delayed`
`end`

## Client Output
 ================================================================
           Error executing action `install` on resource 'windows_package[7zip]'        ================================================================
       
           RuntimeError
           ------------
           **implicit argument passing of super from method defined by define_method() is not supported. Specify all arguments explicitly.**
       
           Resource Declaration:
           ---------------------
           # In C:/Users/azure/AppData/Local/Temp/kitchen/cache/cookbooks/test-recipe/recipes/test_winres.rb
       
             7: windows_package '7zip' do
             8:   source 'http://www.7-zip.org/a/7z938-x64.msi'
             9:   action :install
            10:   notifies :request_reboot, 'reboot[now]', :delayed
            11: end
       
           Compiled Resource:
           ------------------
           # Declared in C:/Users/azure/AppData/Local/Temp/kitchen/cache/cookbooks/test-recipe/recipes/test_winres.rb:7:in `from_file'
       
           windows_package("7zip") do
             package_name "7zip"
             action [:install]
             default_guard_interpreter :default
             declared_type :windows_package
             cookbook_name "p66-ad-join"
             recipe_name "test_winres"
             source "http://www.7-zip.org/a/7z938-x64.msi"
             version "9.38.00.0"
           end

## Stacktrace
Running handlers:
       [2020-04-30T06:22:04+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2020-04-30T06:22:04+00:00] ERROR: Exception handlers complete
       Chef Infra Client failed. 1 resources updated in 08 seconds
       [2020-04-30T06:22:04+00:00] FATAL: Stacktrace dumped to C:/Users/azure/AppData/Local/Temp/kitchen/cache/chef-stacktrace.out
       [2020-04-30T06:22:04+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2020-04-30T06:22:04+00:00] FATAL: RuntimeError: windows_package[7zip] (test-recipe::test_winres line 7) had an error: RuntimeError: implicit argument passing of super from method defined by define_
method() is not supported. Specify all arguments explicitly.

**Comment by tas50** [comment]
@cookiecurse Can you attach the stacktrace file as well?

**Comment by Xorima** [comment]
@tas50 I have replicated this and stacktrace is:

```
Generated at 2020-04-30 18:16:36 +0000
RuntimeError: windows_package[7zip] (test::foobar line 7) had an error: RuntimeError: implicit argument passing of super from method defined by define_method() is not supported. Specify all arguments explicitly.
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/provider/package/windows.rb:150:in `block in <class:Windows>'
(eval):2:in `block in action_install'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/provider.rb:279:in `instance_eval'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/provider.rb:279:in `compile_and_converge_action'
(eval):2:in `action_install'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/provider.rb:220:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource.rb:591:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:74:in `run_action'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:108:in `block in run_all_actions'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:108:in `each'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:108:in `run_all_actions'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:132:in `block in converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/resource_list.rb:96:in `block in execute_each_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:115:in `call_iterator_block'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:86:in `step'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/resource_collection/resource_list.rb:94:in `execute_each_resource'
C:/opscode/chef/embedded/lib/ruby/2.7.0/forwardable.rb:235:in `execute_each_resource'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/runner.rb:130:in `converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/client.rb:714:in `block in converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/client.rb:709:in `catch'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/client.rb:709:in `converge'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/client.rb:733:in `converge_and_save'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/client.rb:288:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/application.rb:292:in `run_with_graceful_exit_option'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/application.rb:268:in `block in run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/local_mode.rb:42:in `with_server_connectivity'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/application.rb:251:in `run_chef_client'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/application/base.rb:335:in `run_application'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.0.257-universal-mingw32/lib/chef/application.rb:68:in `run'
C:/opscode/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-bin-16.0.257/bin/chef-client:25:in `<top (required)>'
C:/opscode/chef/bin/chef-client:184:in `load'
C:/opscode/chef/bin/chef-client:184:in `<main>'
```

**Comment by cookiecurse** [comment]
Thanks @Xorima !

**Comment by tas50** [comment]
Fixed by https://github.com/chef/chef/pull/9789

**Comment by tas50** [comment]
The actual fix went in with #9790. We'll try to get a release out with this fix ASAP. Thanks for the report
